v0/destinations/appcues/transform.js

```diff
@@ -0,0 +1,134 @@
+/* eslint-disable no-underscore-dangle */
+const { EventType } = require("../../../constants");
+
+const {
+  constructPayload,
+  defaultRequestConfig,
+  getValueFromMessage,
+  removeUndefinedAndNullValues,
+  removeUndefinedAndNullAndEmptyValues
+} = require("../../util");
+
+const { ConfigCategory, mappingConfig, getEndpoint } = require("./config");
+
+function buildResponse(payload, endpoint) {
+  const response = defaultRequestConfig();
+  response.endpoint = endpoint;
+  response.body.JSON = removeUndefinedAndNullValues(payload);
+  return {
+    ...response,
+    headers: {
+      "Content-Type": "application/json"
+    }
+  };
+}
+
+function processTrackEvent(message, destination, mappingJson) {
+  const requestJson = constructPayload(
+    message,
+    mappingConfig[ConfigCategory.DEFAULT.name]
+  );
+
+  const eventJson = constructPayload(message, mappingJson);
+  requestJson.events = [eventJson];
+
+  return buildResponse(
+    requestJson,
+    getEndpoint(destination.Config.accountId, message.userId)
+  );
+}
+
+function processPage(
+  message,
+  destination,
+  trackMappingJson,
+  profileMappingJson
+) {
+  const requestJson = constructPayload(
+    message,
+    mappingConfig[ConfigCategory.DEFAULT.name]
+  );
+
+  // generating profile part of the payload
+  const profileJson = constructPayload(message, profileMappingJson);
+  profileJson._appcuesId = destination.Config.accountId;
+  requestJson.profile_update = profileJson;
+
+  // generating event part of the payload
+  const eventJson = constructPayload(message, trackMappingJson);
+  eventJson.attributes = {
+    ...eventJson.attributes,
+    _identity: {
+      userId: message.userId
+    }
+  };
+  eventJson.attributes = removeUndefinedAndNullAndEmptyValues(
+    eventJson.attributes
+  );
+
+  if (
```

Comment:
can we put this in mapping json like
destKey: context.url
sourceKeys: "pageUrl",
sourceFromGenericMap: true